File: operations.java
import java.lang.reflect.Array;

class operations {

    public float[] SubVectors(float[] first, float[] second){
        int size = Array.getLength(first);
        float[] result = new float[size];

        for (int i = 0; i < size; i++) {
            result[i] = first[i] - second[i];
        }

        return result;
    }

    public float MultiplyVectors(float[] first, float[]second){
        int size = Array.getLength(first);
        float result = 0;

        for(int i=0; i<size; i++){
            result+=first[i]*second[i];
        }

        return result;
    }

    public float[] MultiplyMatrixVector(float[][] mat, float[] vec){
        int size = Array.getLength(vec);
        float[] result = new float[size];

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                result[i] += vec[i] * mat[i][j];
            }
        }

        return result;
    }

    public float[][] MultiplyMatrix(float[][] first, float[][] second){
        int size = Array.getLength(first);
        float[][] result = new float[size][size];

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                for (int k = 0; k < size; k++) {

                    result[i][j] += first[i][k] * second[k][j];

                }
            }
        }

        return result;
    }

    public float[][] SumMatrix(float[][] first, float[][] second){
        int size = Array.getLength(first);
        float[][] result = new float[size][size];

        for (int i = 0; i < size; i++) {
            for (int j=0; j<size; j++){
                result[i][j] = first[i][j] + second[i][j];
            }
        }

        return result;
    }

    public float[][]  SortMatrix(float[][] mat){
        int size = Array.getLength(mat);
        
        int check = 1;

        while(check!=0){
            check=0;
            for(int i=0; i<size; i++){
                for(int j=0; j<size-1; j++){
                    if(mat[i][j]<mat[i][j+1]){
                        float memory = mat[i][j];
                        mat[i][j] = mat[i][j+1];
                        mat[i][j+1] = memory;
                        check+=1;
                    }
                }
            }
        }
        return mat;
    }

}
    

File: Python/Operations.py
def SubtractVectors(first, second):
    return [first[i]- second[i] for i in range(len(first))]

def MultiplyVectors(first, second):
    return sum([first[i]*second[i] for i in range(len(first))])

def MultiplyMatrixVector(matrix, vector):
    return [sum([vector[i]*matrix[i][j] for j in range(len(vector))]) for i in range(len(vector))]

def MultiplyMatrix(first, second):
    return [[sum([first[i][k]*second[k][j] for k in range(len(first))]) for j in range(len(first))] for i in range(len(first))]

def SumMatrix(first, second):
    return [[first[i][j] + second[i][j] for j in range(len(first))] for i in range(len(first))]

def SortMatrix(matrix):
    return [sorted([matrix[i][j] for j in range(len(matrix))], reverse = True) for i in range(len(matrix))]